### PR TITLE
Bugfix: Markets data header on front page

### PR DESCRIPTION
**Pull request by adambraimbridge**
@ironsidevsquincy @aintgoin2goa @laurieboyes @andygout

### CODE REVIEW

File: client/components/markets-data/markets-data.js [hunk 1]
```diff
@@ -71,6 +71,10 @@ const createAriaLabel = (priceChangeDirection, percentChange) =>
 	`${priceChangeDirection} ${percentChange.toString().replace(/\./, ' point ')} %`;
 
 export default flags => {
+	const marketsDataEl = document.querySelector('.js-markets-data');
```
Comment (ironsidevsquincy) [inline client/components/markets-data/markets-data.js]: don't need https://github.com/Financial-Times/next-front-page/blob/4b5b0d138e81a7842887d09603f301f4910996fb/client/components/markets-data/markets-data.js#L82 then (surprised linting allowed this)

File: client/components/markets-data/markets-data.js [hunk 2]
```diff
@@ -71,6 +71,10 @@ const createAriaLabel = (priceChangeDirection, percentChange) =>
...
+	if (typeof(marketsDataEl) === "undefined") {
```
Comment (ironsidevsquincy) [inline client/components/markets-data/markets-data.js]: can merge with conditional below

File: server/routes/front-page.js [hunk 3]
```diff
@@ -78,8 +77,15 @@ export default region => (req, res) => {
...
-
+	// If the newHeader flag is active AND the frontPageHeaderMarketsData flag is active,
+	// then show markets data to everyone;
+	// Otherwise, show markets data only to logged-in users.
+	let showMarketsData;
+	if (res.locals.flags.newHeader) {
+		showMarketsData = res.locals.flags.frontPageHeaderMarketsData;
+	} else {
+		showMarketsData = res.locals.anon.userIsLoggedIn;
+	}
```
Comment (ironsidevsquincy) [inline server/routes/front-page.js]: think @andygout might have already something around this

Comment (andygout) [inline server/routes/front-page.js]: Yup: https://github.com/Financial-Times/next-front-page/pull/848

**Comment by adambraimbridge** [comment]
@andygout fixed the original bug already.

**Comment by ironsidevsquincy** [comment]
Think the first bit still useful, checking to see if the element exists before continuing

**Pull request closed without merging**